src/site/content/en/blog/disallow-synchronous-xhr/index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Disallowing synchronous XMLHttpRequest() during page dismissal
+subhead: Reducing delayed navigations
+description: |
+  It's common for a page or app to have unsubmitted analytics or other data at
+  the time a user closes it. Sites use a synchronous call to XMLHttpRequest()
+  to keep the page or app open until its data is passed to the server. It hurts
+  the user experience and ignores better ways to save data. Chrome 80 implements
+  a recent spec change to address this
+authors:
+  - joemedley
+date: 2019-12-18
+hero: hero.jpg
+alt: Roadblock barricades
+tags:
+  - post
+  - deprecation
+---
+
+It's common for a page or app to have unsubmitted analytics or other data at the
+time a user closes it. To prevent data loss, some sites use a synchronous call
+to `XMLHttpRequest()` to keep the page or app open until its data is passed to
+the server. Not only are there better ways to save data, but this technique creates
+a bad user experience by delaying closing of the page for up to several seconds.
+
+This practice needs to change, and browsers are responding. The `XMLHttpRequest()`
+specification is already [slated for deprecation and
+removal](https://xhr.spec.whatwg.org/#sync-warning). Chrome 80 takes the first
+step by disallowing synchronous calls inside several event handlers,
+specifically `beforeunload`, `unload`, `pagehide`, and `visibilitychange` when
+they are fired in the dismissal. WebKit also recently landed [a commit implementing
+the same behavior change](https://bugs.webkit.org/show_bug.cgi?id=204912).
+
+In this article I'll briefly describe options for those who need time to update
+their sites and outline the alternatives to `XMLHttpRequest()`.
+
+## Temporary opt-outs
+
+Chrome does not simply want to pull the plug on `XMLHttpRequest()`, which is why a few
+temporary opt-out options are available. For sites on the internet, [an origin
+trial is
+available](https://developers.chrome.com/origintrials/#/view_trial/4391009636686233601).
+With this, you add an origin-specific token to your page headers that enables
+synchronous `XMLHttpRequest()` calls. This option ends shortly before Chrome 86
+ships, sometime in late October of 2020. Enterprise Chrome customers can also
+use the  `AllowSyncXHRInPageDismissal` policy flag, which ends at the same time.
+
+## Alternatives
+
+Regardless of how you send data back to the server, it's best to avoid waiting
+until page unload to send all the data at once. Aside from creating a bad user
+experience, you risk data loss if something goes wrong.  Unload events [often
+don't fire on mobile
+browsers](https://www.igvita.com/2015/11/20/dont-lose-user-and-app-state-use-page-visibility/)
+because there are [many ways to
+close](https://developers.google.com/web/updates/2018/07/page-lifecycle-api) a
+tab or browser on mobile operating systems without the unload event firing. With
```

Comment:
```suggestion
tab or browser on mobile operating systems without the `unload` event firing. With
```

If we're talking about unload events as a general category of related events then you can ignore this. If we're specifically talking about the `unload` event that the browser fires then I think we should format it as inline code.